### ISSUE TITLE
Music only comes out of left channel

### ISSUE THREAD

**Issue by mlindner** [arescentral/antares#216]
This has been an outstanding issue since at least version 0.7.2, but didn't exist in 0.6.1. When launching the game the opening music, or any in-game music only plays from the left channel. Sound effects still come out of both channels.

Mac OS X 10.10 with integrated and discrete GPU.

**Comment by sfiera** [comment]
This would probably date to 0.7.0. Prior to that release, Antares used MP3s of the music. Now, it uses the music in roughly the original format (MADH/S3M). Going to do some investigation here after a restart.

**Comment by sfiera** [comment]
The right channel sounds quieter to me, but not silent. I got the libmodplug output into Audacity and that definitely checks out:
<img width="1052" alt="screen shot 2018-05-06 at 11 01 33" src="https://user-images.githubusercontent.com/166772/39671668-0ae475a2-511d-11e8-9e55-c33fecbab76d.png">
Weird.

I’ll dig into this some more.

**Comment by mlindner** [comment]
I believe discovered the origin of the problem. I assume that you used PlayerPRO Player.app to export the MADH file into an s3m file. If you play the s3m file it generates in PlayerPRO Player it plays correctly, however if you play it in any other mod file player (tried both libmodplug and mikmod/libmikmod) they play it incorrectly. Either both of those libraries implement s3m files incorrectly OR PlayerPRO is generating the format incorrectly.

**Comment by sfiera** [comment]
Interesting. I had tried with MilkyTracker, which seems to plays the S3Ms as if they were monophonic (I wasn’t aware that channels were locked). I used PlayerPRO for the conversion, yes.

One feature of libmodplug is “stereo separation”. If I set `settings.mStereoSeparation = 256` in openal-driver.cpp, then the right channel is completely absent for me. If I set it to 0, then I get monophonic music.

My working theory right now would be:
1. Somehow `mStereoSeparation` is 256 for you and in the middle for me.
2. PlayerPRO is assuming MOD channels are locked by index, and zeroes out some value. But they aren’t locked in S3M, and conforming S3M players assume zero to mean left channel.

**Comment by sfiera** [comment]
Found it. At offset $00000040, the channels are listed as $01020304 (in all files) which maps all channels to the left speaker. They should be $00080109, mapping to alternate left/right channels.

With headphones, I find 256 stereo separation too harsh. 192 seems tolerable, but still keeps the channels very distinct. I’ll explicitly set that, then.

**Comment by mlindner** [comment]
I would try to leave it how the original game rendered it rather than trying to modify it. Arguably one of the features of MOD-style music is its "harshness", so please keep it that way.

**Comment by mlindner** [comment]
I did some more digging for this. I had a copy of SheepShaver lying around and installed PlayerPro and Ares 1.2.0 into it. If I open up the .madh files (extracted from the resource forks) in PlayerPro, the mixer settings show the panning settings for each channel.

* Channels 1 and 3 have panning set to -50%
* Channels 2 and 4 have panning set to 50%
(values go from -100% (full left) to 100% (full right)

However I discovered these are just the defaults that it loads MADH files at because MADH files have no panning settings (if you try to save them, it saves them as MADK files).

Secondly, the music is VERY different between how its played in PlayerPro and Ares. This is surprising. I think the MADLibrary settings as configured by Ares are different than the file defaults. 

Most importantly it completely ignores the panning settings I mentioned above, it plays the MADH files as channel-locked with full stereo separation.
